Userland/Applications/PixelPaint/ImageEditor.cpp

```diff
@@ -92,6 +92,8 @@ bool ImageEditor::undo()
     m_undo_stack.undo();
     m_undo_stack.redo();
     layers_did_change();
+
+    window()->set_modified(is_modified());
```

Comment:
this should also be in `redo()`, the modified state is not set correctly if you save -> undo -> redo

Reply:
Totally right. I even looked at redo() and somehow reasoned to myself it didn't need it lol. I'll fix that tomorrow and get the change pushed.  Thanks for catching that

Reply:
I've gone ahead and dropped this change all together as I see your PR #15919 handles this modified status a bit better. This PR now only covers the Tool -> Image Editor sync problem.